nucypher/policy/conditions/json/rpc.py

```diff
@@ -0,0 +1,177 @@
+
+
+class JsonEndpointRPCCall(BaseJsonRPCCall):
+    class Schema(BaseJsonRPCCall.Schema):
+        endpoint = Url(required=True, relative=False, schemes=["https"])
+
+        @post_load
+        def make(self, data, **kwargs):
+            return JsonEndpointRPCCall(**data)
+
+    def __init__(
+        self,
+        endpoint: str,
+        method: str,
+        params: Optional[Any] = None,
+        query: Optional[str] = None,
+        authorization_token: Optional[str] = None,
+    ):
+        self.endpoint = endpoint
+        super().__init__(
+            method=method,
+            params=params,
+            query=query,
+            authorization_token=authorization_token,
+        )
+
+    @override
+    def execute(self, **context) -> Any:
+        return super()._execute(endpoint=self.endpoint, **context)
+
+
+class JsonRpcCondition(ExecutionCallAccessControlCondition):
+    EXECUTION_CALL_TYPE = JsonEndpointRPCCall
+    CONDITION_TYPE = ConditionType.JSONRPC.value
+
+    class Schema(
+        ExecutionCallAccessControlCondition.Schema, JsonEndpointRPCCall.Schema
+    ):
+        condition_type = fields.Str(
+            validate=validate.Equal(ConditionType.JSONRPC.value), required=True
+        )
+
+        @post_load
+        def make(self, data, **kwargs):
+            return JsonRpcCondition(**data)
+
+    def __init__(
+        self,
+        endpoint: str,
+        method: str,
+        return_value_test: ReturnValueTest,
+        params: Optional[Any] = None,
+        query: Optional[str] = None,
+        authorization_token: Optional[str] = None,
+        condition_type: Optional[str] = ConditionType.JSONRPC.value,
+        name: Optional[str] = None,
+    ):
+        super().__init__(
+            endpoint=endpoint,
+            method=method,
+            return_value_test=return_value_test,
+            params=params,
+            query=query,
+            authorization_token=authorization_token,
+            condition_type=condition_type,
+            name=name,
+        )
+
+    @property
+    def endpoint(self):
+        return self.execution_call.endpoint
+
+    @property
+    def method(self):
+        return self.execution_call.method
+
+    @property
+    def params(self):
+        return self.execution_call.params
+
+    @property
+    def query(self):
+        return self.execution_call.query
+
+    @property
+    def authorization_token(self):
+        return self.execution_call.authorization_token
+
+    @property
+    def timeout(self):
+        return self.execution_call.timeout
+
+    def verify(self, **context) -> Tuple[bool, Any]:
+        result = self.execution_call.execute(**context)
+        result_for_eval = process_result_for_condition_eval(result)
+
+        resolved_return_value_test = self.return_value_test.with_resolved_context(
+            **context
+        )
+        eval_result = resolved_return_value_test.eval(result_for_eval)  # test
+        return eval_result, result
```

Comment:
Code for `verify()` method seems identical to `JsonApiCondition.verify()`. I'm wondering if there's a way to generalize both. If I understand it correctly, on an abstract level, `JsonRpcCondition` is just a subcase of `JsonApiCondition`, even if it's not explicitly inheriting from it now.

Reply:
✔️ - it is indeed identical.

It's subtle, but JsonRpcCondition can't subclass JsonApiCondition. JsonApiCondition only allows HTTP GET, and JsonRpcCondition only allows HTTP POST. Also `JsonRpcCondition` takes `method` which `JsonApiCondition` doesn't.

Created a common base superclass instead (similar to what I did for the `JsonApiCall` and `JsonRpcCall`).